scripts/docker.sh

```diff
@@ -0,0 +1,75 @@
+#! /bin/bash
+#
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+repo_base=$(cd "$(dirname "$(dirname "$0")")" || exit ; pwd)
+
+UNSTABLE_TAG=unstable
+DEFAULT_GCR=${DEFAULT_GCR:-gcr.io/kpt-fn-contrib}
+GCR_REGISTRY=${GCR_REGISTRY:-${DEFAULT_GCR}}
+
+function err {
+  echo "$1"
+  exit 1
+}
+
+function docker_build {
+  type=$1 # function type, e.g. contrib, curated
+  lang=$2 # function language, e.g. go, ts
+  name=$3 # function name, e.g. apply-setters
+
+  build_args=()
+
+  case "${type}" in
+    contrib) function_dir="${repo_base}/contrib/functions/${lang}/${name}" ;;
+    curated) function_dir="${repo_base}/functions/${lang}/${name}" ;;
+    *) err "unknown function type: ${type}" ;;
+  esac
+
+  case "${lang}" in
+    ts)
+      translated_name=$(echo "${name}" | tr - _)
```

Comment:
We won't need to do it if we address https://github.com/GoogleContainerTools/kpt/issues/2533.
What is the status of https://github.com/GoogleContainerTools/kpt/issues/2533?

Reply:
I have not started on 2533 yet, although I do agree that resolving 2533 would simplify this logic

Reply:
Got it